rcl/test/rcl/test_security.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <string>
+
+#include "rcl/security.h"
+#include "rcl/error_handling.h"
+
+#include "rcutils/filesystem.h"
+
+#include "rmw/error_handling.h"
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+
+
+#define TEST_SECURITY_DIRECTORY_RESOURCES_DIR_NAME "/test_security_directory"
+#define TEST_SECURITY_CONTEXT "dummy_security_context"
+#define TEST_SECURITY_CONTEXT_ABSOLUTE "/" TEST_SECURITY_CONTEXT
+
+#ifndef _WIN32
+# define PATH_SEPARATOR "/"
+#else
+# define PATH_SEPARATOR "\\"
+#endif
+
+char g_envstring[512] = {0};
+
+static int putenv_wrapper(const char * env_var)
+{
+#ifdef _WIN32
+  return _putenv(env_var);
+#else
+  return putenv(reinterpret_cast<char *>(const_cast<char *>(env_var)));
```

Comment:
Hmm, why an (apparently redundant) `reinterpret_cast` here?

Reply:
good question, it's preexistence but I will update.

Reply:
See https://github.com/ros2/rcl/pull/515/commits/e3d5ebdf6094274f7f74d752ecce3fbadec84150.